sxg_rs/src/acme/client.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An ACME client handles authentication with the ACME server.
+
+use super::directory::Directory;
+use crate::crypto::EcPublicKey;
+use crate::fetcher::Fetcher;
+use crate::http::{HttpRequest, HttpResponse, Method};
+use crate::signature::Signer;
+use anyhow::{anyhow, Result};
+use serde::Serialize;
+
+pub struct Client<F: Fetcher, S: Signer> {
+    pub public_key: EcPublicKey,
+    pub directory: Directory,
+    nonce: Option<String>,
+    fetcher: F,
+    signer: S,
+}
+
+#[derive(PartialEq, Eq)]
+pub enum AuthMethod<'a> {
+    JsonWebKey,
+    KeyId(&'a str),
+}
+
+impl<F: Fetcher, S: Signer> Client<F, S> {
```

Comment:
Please add unit tests.

Reply:
Done. The unit test is added in `sxg_rs/src/acme/mod.rs`.